nil pointer dereference when running `minikube addons configure registry-creds`

### What Happened?

```
henry@DESKTOP-KILRQMD:~/oss/21685/minikube$ minikube addons configure registry-creds --v=8 --alsologtostderr
I1022 02:28:32.091906   47501 out.go:360] Setting OutFile to fd 1 ...
I1022 02:28:32.092464   47501 out.go:413] isatty.IsTerminal(1) = true
I1022 02:28:32.092523   47501 out.go:374] Setting ErrFile to fd 2...
I1022 02:28:32.092564   47501 out.go:413] isatty.IsTerminal(2) = true
I1022 02:28:32.092902   47501 root.go:338] Updating PATH: /home/henry/.minikube/bin
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x29f8675]

goroutine 1 [running]:
k8s.io/minikube/cmd/minikube/cmd/config.processRegistryCredsConfig({0x353aa93, 0x8}, 0x0)
        /app/cmd/minikube/cmd/config/configure_registry_creds.go:93 +0x35
k8s.io/minikube/cmd/minikube/cmd/config.init.func8(0xc00078ee00?, {0xc000d81e90, 0x1, 0x35326d3?})
        /app/cmd/minikube/cmd/config/configure.go:69 +0x21e
github.com/spf13/cobra.(*Command).execute(0x6306720, {0xc000d81d40, 0x3, 0x3})
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x6302260)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
k8s.io/minikube/cmd/minikube/cmd.Execute()
        /app/cmd/minikube/cmd/root.go:174 +0x591
main.main()
        /app/cmd/minikube/main.go:95 +0x235
```

### Attach the log file

[log.txt](https://github.com/user-attachments/files/23059880/log.txt)

### Operating System

Ubuntu

### Driver

Docker

--- a/cmd/minikube/cmd/config/configure.go
+++ b/cmd/minikube/cmd/config/configure.go
@@ -156,10 +156,9 @@ func loadAddonConfigFile(addon, configFilePath string) (ac *addonConfig) {
 			exit.Message(reason.Kind{ExitCode: reason.ExProgramConfig, Advice: "provide a valid config file"},
 				fmt.Sprintf("error reading config file: %v", err))
 		}
-
-		return &cf.Addons
 	}
-	return nil
+
+	return &cf.Addons
 }
 
 // Processes metallb addon config from configFile if it exists otherwise resorts to default behavior

